int32 start
int32 end
---
string result